pkg/observability/modules/stackstate.ts

```diff
@@ -183,3 +194,49 @@ export function loadComponent(
 function token(apiToken: string, serviceToken: string): string {
   return apiToken ? `ApiToken ${ apiToken }` : `ApiKey ${ serviceToken }`;
 }
+
+/**
+ * Ensure that there is a NodeDriver called 'stackstate' that has the URL whitelisted, so that the
+ * Metadata Proxy can call out to the StackState API.
+ *
+ * NOTE: Be aware that this goes through Norman APIs, not the Steve ones.
+ *
+ * @param store
+ * @param url
+ * @returns
+ */
+export async function ensureObservabilityUrlWhitelisted(store: any, url: string): Promise<boolean> {
+  const nodeDrivers = await store.dispatch('rancher/findAll', { type: 'nodeDriver' }, { root: true });
+
+  let stackStateDriver;
+
+  if (!nodeDrivers || isEmpty(nodeDrivers)) {
+    // Need to create new
+  } else {
+    stackStateDriver = nodeDrivers.find((driver: any) => driver.name === 'stackstate');
+  }
```

Comment:
I think all of the block from 211 to 217 could be just:

```
let stackStateDriver = = nodeDrivers.find((driver: any) => driver.name === 'stackstate');
```

`nodeDrivers` should be an empty array, even if there are no node drivers.

Also because of JS liking to keep object references I don't recommend re-using these type of variables... With that in mind I would do something like:

```
const stackStateDriver = = nodeDrivers.find((driver: any) => driver.name === 'stackstate');
```

and adjust the code below this.

I've spent too much time of my life debugging problems with object references and I just avoid them from the start, if I can 😛 

Feel free to ignore this `const` comment (and the subsequent I did below) if everything is working fine 😄